src/driver/mongodb/MongoDriver.ts

```diff
@@ -442,21 +442,22 @@ export class MongoDriver implements Driver {
     /**
      * Builds connection url that is passed to underlying driver to perform connection to the mongodb database.
      */
-    protected buildConnectionUrl(options: { [key: string]: any }): string {
+    static buildConnectionUrl(options: { [key: string]: any }): string {
```

Comment:
Can this still be protected even if it's static?  I guess not because that's how you're confirming this with tests?

Is there any way to test that this change works WITHOUT exposing more functions as part of a public API?

Reply:
Yeah we thought about this two. The unfortunate part is that the one public method using this is the connect method - so then we'd have to mock the world to get a test. But I wanted to have a test 🙈

Reply:
Ok now I rewrote the test a bit - don't know whether it's really pretty now. But what it does now is mock the connect function and emulate what the mongodriver does, so we can actually catch the connection url and verify that it is correct. A bit hard to read, but that way there is no additional public method.